src/components/Inventory/withdraw/flow/SelectBatchesWithdrawnQuantity.tsx

```diff
@@ -1,20 +1,160 @@
-import React from 'react';
-import { Typography } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { Container, Grid, Typography, useTheme } from '@mui/material';
 import { ServerOrganization } from 'src/types/Organization';
 import FormBottomBar from 'src/components/common/FormBottomBar';
+import { useDeviceInfo } from '@terraware/web-components/utils';
+import strings from 'src/strings';
+import { NurseryWithdrawalRequest } from 'src/api/types/batch';
+import { Table, TableColumnType } from '@terraware/web-components';
+import WithdrawalBatchesCellRenderer from './WithdrawalBatchesCellRenderer';
+import useForm from 'src/utils/useForm';
 
 type SelectBatchesWithdrawnQuantityProps = {
   organization: ServerOrganization;
   onNext: () => void;
   onCancel: () => void;
   saveText: string;
+  batches: any[];
+  nurseryWithdrawal: NurseryWithdrawalRequest;
 };
+
+type BatchWithdrawalForTable = {
+  batchId: number;
+  batchNumber: string;
+  notReadyQuantityWithdrawn: number;
+  readyQuantityWithdrawn: number;
+  notReadyQuantity: number;
+  readyQuantity: number;
+  speciesId: number;
+  facilityName: string;
+};
+
 export default function SelectBatches(props: SelectBatchesWithdrawnQuantityProps): JSX.Element {
-  const { onNext, onCancel, saveText } = props;
+  const { onNext, onCancel, saveText, batches, nurseryWithdrawal, organization } = props;
+  const theme = useTheme();
+  const { isMobile } = useDeviceInfo();
+  const [species, setSpecies] = useState<any>([]);
+  const [record, setRecord] = useForm<BatchWithdrawalForTable[]>([]);
+
+  useEffect(() => {
+    const transformBatchesForTable = async () => {
+      const speciesFromBatches: { [x: string]: { id: number; scientificName: string; commonName: string } } = {};
+      let batchesForTable: BatchWithdrawalForTable[] = [];
+      batchesForTable = nurseryWithdrawal.batchWithdrawals.reduce((acc, bw) => {
```

Comment:
can combine these 2
const batchesForTable: BatchWithdrawalForTable[] = nurseryWithdrawal.batchWithdawals.reduce

Reply:
I try doing this but at the end of the reduce function, when setting the initValue, I get `Variable 'batchesForTable' is used before being assigned.` and if I don't set the init value typescript thinks the accumulator is of type BatchWithdrawal instead of BatchWithdrawalForTable

Reply:
We can initialize it as `[]`;
```
const batchesForTable: BatchWithdrawalForTable[] = nurseryWithdarwal.batchWitdhrawals.reduce((...) => {
   ...
}, []);
```

Reply:
I get a typescript error: `Argument of type '{ batchId: ... }' is not assignable to parameter of type 'never'`

Reply:
ugh.
this will work 
```
const batchesForTable: BatchWithdrawalForTable[] = nurseryWithdarwal.batchWitdhrawals.reduce((...) => { 
    ... 
}, [] as BatchWithdrawalForTable[]);
```
Will be cleaner this way than splitting into 2 lines just for typescript sake